### PR TITLE
Removed subprocess and some small script improvements :)

### PR DESCRIPTION
**Pull request by yuannan**

### CODE REVIEW

File: panon/backend/source.py [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import numpy as np
+import soundcard as sc
```
Comment (rbn42) [inline panon/backend/source.py]: I didn't expect everyone would be using PulseAudio. They can use PortAudio as a fallback. So import soundcard may cause problems for them.

Comment (yuannan) [reply panon/backend/source.py]: ah okay, that's why this was within the body.

I did this for performance reasons as imports during runtime. If an item is imported more than twice it doesn't bug out but it does take up CPU.

Feel free to revert the changes to this.

Comment (rbn42) [reply panon/backend/source.py]: >  If an item is imported more than twice it doesn't bug out but it does take up CPU.

Really? If so, you can put the statement in the head, within try except blocks.
```python
try:
   import soundcard
except:
   pass
```

**Comment by yuannan** [comment]
reverted :)

**Comment by rbn42** [comment]
https://github.com/rbn42/panon/blob/864dd15bd4800cb461054286398d537bf1b4c5e4/plasmoid/contents/ui/config/ConfigBackend.qml#L189

So "Any Active Speaker" is no longer a suitable name. Do you have a better suggestion?

**Comment by yuannan** [comment]
> https://github.com/rbn42/panon/blob/864dd15bd4800cb461054286398d537bf1b4c5e4/plasmoid/contents/ui/config/ConfigBackend.qml#L189
> 
> So "Any Active Speaker" is no longer a suitable name. Do you have a better suggestion?

"Monitor Active Speaker"? or "Monitor of Current Speaker" or "Monitor of Default Speaker"
Fits in the current options better :)
I think "Monitor of Default Speaker" is the most accurate option but "Monitor of Current Speaker" is more user friendly.
image

**Comment by rbn42** [comment]
> Monitor Active Speaker
> = p.server_info['default sink id']

Your code is searching for the default one, I mean, not an active one. 

And unlike the original "default device", you assumed the default device could be changed during running.

I would come to a name like "Hotplugging default device". I hope you have a better idea.

**Comment by yuannan** [comment]
> > Monitor Active Speaker
> > = p.server_info['default sink id']
> 
> Your code is searching for the default one, I mean, not an active one.
> 
> And unlike the original "default device", you assumed the default device could be changed during running.

It's fine. Pulse keeps this up to date and soundcard has multi-processing support with locks :). It's not the first device but instead the default CURRENT playback device. It works just fine on my setup when I switch between them.

I'm pretty sure under KDE when you change your device it changes all the sources to the new sink.

AFIAK, this completes the exact same function as the old code but just faster as it uses a native python library that calls C libs instead of using a subprocess.

**Comment by rbn42** [comment]
Everything is good, except for the name.

Do you want me to merge now? Or do you want to rename the "Any Active Speaker" option?

**Comment by yuannan** [comment]
I think "Current Playback Device" would be my vote right now. As this implies that it's needed to be hot plugged which is not true. We don't need to expose the user to the implementation of the code.

**Comment by yuannan** [comment]
Change it to "Monitor Current Device" and merge it please :)

**Pull request merged**